go.mod

```diff
@@ -1,100 +1,64 @@
-	k8s.io/apimachinery v0.0.0-20181128191346-49ce2735e507
-	k8s.io/apiserver v0.0.0-20181207191401-9601a7bf41ef
-	k8s.io/client-go v0.0.0-20181205110444-5e6a3d4e34f6
-	k8s.io/csi-api v0.0.0-20181121073239-b5848c3064d4 // indirect
-	k8s.io/kube-controller-manager v0.0.0-20181110193944-1c4e86102515 // indirect
-	k8s.io/kube-openapi v0.0.0-20180711000925-0cf8f7e6ed1d // indirect
-	k8s.io/kubernetes v1.12.4
-	k8s.io/utils v0.0.0-20180726175726-66066c83e385 // indirect
-	kubevirt.io/containerized-data-importer v1.5.1 // indirect
-	kubevirt.io/kubevirt v0.16.0
+	github.com/golang/mock v1.4.3
+	github.com/spf13/cobra v1.0.0
+	github.com/spf13/pflag v1.0.5
+	gopkg.in/yaml.v2 v2.3.0
+	k8s.io/api v0.17.1
+	k8s.io/apimachinery v0.17.1
+	k8s.io/apiserver v0.16.4
+	k8s.io/client-go v12.0.0+incompatible
+	k8s.io/cloud-provider v0.16.4
+	k8s.io/component-base v0.16.4
+	k8s.io/kubernetes v1.16.4
+	kubevirt.io/client-go v0.29.2
+)
+
+replace (
+	github.com/go-kit/kit => github.com/go-kit/kit v0.3.0
+	github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.3.1
+	github.com/prometheus/client_golang => github.com/prometheus/client_golang v0.9.2
+	k8s.io/api => k8s.io/api v0.16.4
+	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.16.4
+	k8s.io/apimachinery => k8s.io/apimachinery v0.16.4
+	k8s.io/apiserver => k8s.io/apiserver v0.16.4
+	k8s.io/cli-runtime => k8s.io/cli-runtime v0.16.4
+	k8s.io/client-go => k8s.io/client-go v0.16.4
+	k8s.io/cloud-provider => k8s.io/cloud-provider v0.16.4
+	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.16.4
+	k8s.io/code-generator => k8s.io/code-generator v0.16.4
+	k8s.io/component-base => k8s.io/component-base v0.16.4
+	k8s.io/cri-api => k8s.io/cri-api v0.16.4
+	k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.16.4
+	k8s.io/klog => k8s.io/klog v0.4.0
+	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.16.4
+	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.16.4
+	k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20191107075043-30be4d16710a
+	k8s.io/kube-proxy => k8s.io/kube-proxy v0.16.4
+	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.16.4
+	k8s.io/kubectl => k8s.io/kubectl v0.0.0-20200124035537-9f7d91504e51
+	k8s.io/kubelet => k8s.io/kubelet v0.16.4
+	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.16.4
+	k8s.io/metrics => k8s.io/metrics v0.16.4
+	k8s.io/node-api => k8s.io/node-api v0.16.4
+	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.16.4
+	k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.16.4
+	k8s.io/sample-controller => k8s.io/sample-controller v0.16.4
+)
+
+// sigs.k8s.io/controller-runtime 0.6.0
+// and
+// github.com/openshift/machine-api-operator v0.2.1-0.20200402110321-4f3602b96da3 (throgh its import github.com/openshift/client-go)
+// requires k8s-* v0.18.2
+// but we are pinned to kubernetes-1.16.4 as for kubevirt.io/kubevirt v0.29.1 while we explicitly
+// need github.com/operator-framework/api v0.3.5
+replace (
```

Comment:
Can we union those two replace sections? or there is a reason for that separation?

Reply:
The second replace is done because of kubevirt version limitation
I separated it in order to make it clear what need to be changed when fixing kubevirt version issue
In cluster-api-provider-kubevirt done the same

Reply:
It sound right, ignore my comment